### PR TITLE
SPARKC-138: Added a way to run integration tests on vanilla Spark

### PR DESCRIPTION
**Pull request by jacek-lewandowski**

### CODE REVIEW

**Comment by helena** [comment]
@jacek-lewandowski looks good but I can't test it right now.

**Comment by pkolaczk** [comment]
I tested it, and it works. I had a few tests failing:
{noformat}
[info] - should write values of user-defined classes *** FAILED *** (88 milliseconds)
[info]   "[CustomerId(foo)]" did not equal "[foo]" (TableWriterSpec.scala:270)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.MatchersHelper$.newTestFailedException(MatchersHelper.scala:160)

[info] ActorStreamingSpec:
[info] actorStream
[info] - must write from the actor stream to cassandra table: demo.streaming_wordcount *** FAILED *** (10 seconds, 831 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (10 seconds) during expectMsg:
[info]   at scala.Predef$.assert(Predef.scala:179)
[info]   at akka.testkit.TestKitBase$class.expectMsgPF(TestKit.scala:356)
[info]   at akka.testkit.TestKit.expectMsgPF(TestKit.scala:718)
[info]   at com.datastax.spark.connector.streaming.ActorStreamingSpec$$anonfun$1$$anonfun$apply$mcV$sp$4.apply$mcV$sp(ActorStreamSpec.scala:51)
[info]   at com.datastax.spark.connector.streaming.ActorStreamingSpec$$anonfun$1$$anonfun$apply$mcV$sp$4.apply(ActorStreamSpec.scala:37)
[info]   at com.datastax.spark.connector.streaming.ActorStreamingSpec$$anonfun$1$$anonfun$apply$mcV$sp$4.apply(ActorStreamSpec.scala:37)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
{noformat}

I ran the tests with `dev/run-real-tests.sh 1.2.1 2.10`.
I'll look into the TableWriter error.

**Comment by pkolaczk** [comment]
I verified the TableWriter failing test is fixed by:
https://datastax-oss.atlassian.net/browse/SPARKC-157
https://github.com/datastax/spark-cassandra-connector/pull/664

**Comment by RussellSpitzer** [comment]
Removing "Guava" for the assembly jar breaks the ability to run with the spark shell

    java.lang.NoClassDefFoundError: com/google/common/collect/ImmutableMap
    	at com.datastax.driver.core.ProtocolVersion.<clinit>(ProtocolVersion.java:71)
    	at com.datastax.driver.core.ProtocolOptions.<clinit>(ProtocolOptions.java:76)
    	at com.datastax.driver.core.Cluster$Builder.getConfiguration(Cluster.java:1087)
    	at com.datastax.driver.core.Cluster.<init>(Cluster.java:108)

When run using --jars

    ~/repos/spark/bin/spark-shell --jars ~/repos/spark-cassandra-connector/spark-cassandra-connector/target/scala-2.10/spark-cassandra-connector-assembly-1.3.0-SNAPSHOT.jar --conf spark.cassandra.connection.host=127.0.0.1 --master local

File: project/CassandraSparkBuild.scala [hunk 1]
```diff
@@ -128,7 +137,7 @@ object Dependencies {
     val cassandraDriver     = "com.datastax.cassandra"  % "cassandra-driver-core"  % CassandraDriver guavaExclude // ApacheV2
     val commonsLang3        = "org.apache.commons"      % "commons-lang3"          % CommonsLang3                 // ApacheV2
     val config              = "com.typesafe"            % "config"                 % Config         % "provided"  // ApacheV2
-    val guava               = "com.google.guava"        % "guava"                  % Guava
+    val guava               = "com.google.guava"        % "guava"                  % Guava          % "provided"
```
Comment (RussellSpitzer) [inline project/CassandraSparkBuild.scala]: I'm not sure we want to do this for ease of use with apache spark builds

**Pull request merged**